src/convertToObject.js

```diff
@@ -6,7 +6,18 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styleRules = sourceString.replace(/;\s*;/g, ' ').trim().split(';');
+  const stylesObject = {};
+
+  styleRules.forEach((element) => {
+    const [key, value] = element.split(':');
+
+    if (value !== undefined) {
+      stylesObject[key.trim()] = value.trim();
+    }
+  });
```

Comment:
[CODE PRACTICE] - Avoid using loops in this task, use iteration methods instead. While `forEach` is an exception to the rule against modifying existing arrays/objects, consider using `reduce` to build your `stylesObject` directly from `styleRules`. This will result in a more functional approach and cleaner code.